packages/provider-utils/src/convert-audio-input.ts

```diff
@@ -0,0 +1,218 @@
+import { TypeValidationError } from '@ai-sdk/provider';
+
+/**
+ * Converts various audio input formats to a standardized interface
+ * that provides methods to convert to different audio representations.
+ *
+ * @param audio - The audio input in various formats
+ * @returns An object with methods to convert to different audio formats
+ */
+export function convertAudioInput(
+  audio: Blob | File | Uint8Array | ArrayBuffer | Buffer | URL | string,
+) {
+  // Helper to infer MIME type from filename
+  const inferMimeType = (filename: string): string => {
+    const extension = filename.split('.').pop()?.toLowerCase();
+    switch (extension) {
+      case 'mp3':
+        return 'audio/mpeg';
+      case 'wav':
+        return 'audio/wav';
+      case 'ogg':
+        return 'audio/ogg';
+      case 'flac':
+        return 'audio/flac';
+      case 'm4a':
+        return 'audio/mp4';
+      default:
+        return 'audio/wav'; // Default fallback
+    }
+  };
```

Comment:
would prefer `detect-audio-mimetype` helper that matches `detect-image-mimetype` in same dir as `detect-image-mimetype` (we have optimizations for detecting w/o any conversion)

Reply:
Extract a generalized detect mimetype potentially